### PR TITLE
Simplify UpdateSubTree exception output

### PR DESCRIPTION
**Pull request by smoogipoo**
Because the relevant exceptions [in here](https://ci.appveyor.com/project/peppy/osu/build/master-5904) are unreadable.

From:
```
[ERROR] FATAL UNHANDLED EXCEPTION: System.Exception: Find me!
  at osu.Game.Rulesets.Osu.Objects.Drawables.Pieces.CirclePiece.OnPressed (osu.Game.Rulesets.Osu.OsuAction action) [0x00001] in /Users/smoogipoo/Repos/Temp/temp-osu/osu.Game.Rulesets.Osu/Objects/Drawables/Pieces/CirclePiece.cs:54 
  at osu.Framework.Input.Bindings.KeyBindingContainer`1+<>c__DisplayClass19_1[T].<PropagatePressed>b__1 (osu.Framework.Input.Bindings.IKeyBindingHandler`1[T] d) [0x00000] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Input/Bindings/KeyBindingContainer.cs:156 
  at System.Linq.Enumerable.TryGetFirst[TSource] (System.Collections.Generic.IEnumerable`1[T] source, System.Func`2[T,TResult] predicate, System.Boolean& found) [0x0003f] in /Users/builder/data/lanes/4992/mono-mac-sdk/external/bockbuild/builds/mono-x64/external/corefx/src/System.Linq/src/System/Linq/First.cs:107 
  at System.Linq.Enumerable.FirstOrDefault[TSource] (System.Collections.Generic.IEnumerable`1[T] source, System.Func`2[T,TResult] predicate) [0x00000] in /Users/builder/data/lanes/4992/mono-mac-sdk/external/bockbuild/builds/mono-x64/external/corefx/src/System.Linq/src/System/Linq/First.cs:46 
  at osu.Framework.Input.Bindings.KeyBindingContainer`1[T].PropagatePressed (System.Collections.Generic.IEnumerable`1[T] drawables, T pressed) [0x000bb] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Input/Bindings/KeyBindingContainer.cs:156 
  at osu.Framework.Input.Bindings.KeyBindingContainer`1[T].handleNewPressed (osu.Framework.Input.InputState state, osu.Framework.Input.Bindings.InputKey newKey, System.Boolean repeat) [0x000c8] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Input/Bindings/KeyBindingContainer.cs:129 
  at osu.Framework.Input.Bindings.KeyBindingContainer`1[T].OnMouseDown (osu.Framework.Input.InputState state, osu.Framework.Input.MouseDownEventArgs args) [0x00000] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Input/Bindings/KeyBindingContainer.cs:88 
  at osu.Framework.Graphics.Drawable.TriggerOnMouseDown (osu.Framework.Input.InputState screenSpaceState, osu.Framework.Input.MouseDownEventArgs args) [0x00000] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Graphics/Drawable.cs:1662 
  at osu.Framework.Input.InputManager+<>c__DisplayClass51_0.<PropagateMouseDown>b__0 (osu.Framework.Graphics.Drawable target) [0x00000] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Input/InputManager.cs:552 
  at System.Linq.Enumerable.TryGetFirst[TSource] (System.Collections.Generic.IEnumerable`1[T] source, System.Func`2[T,TResult] predicate, System.Boolean& found) [0x0003f] in /Users/builder/data/lanes/4992/mono-mac-sdk/external/bockbuild/builds/mono-x64/external/corefx/src/System.Linq/src/System/Linq/First.cs:107 
  at System.Linq.Enumerable.FirstOrDefault[TSource] (System.Collections.Generic.IEnumerable`1[T] source, System.Func`2[T,TResult] predicate) [0x00000] in /Users/builder/data/lanes/4992/mono-mac-sdk/external/bockbuild/builds/mono-x64/external/corefx/src/System.Linq/src/System/Linq/First.cs:46 
  at osu.Framework.Input.InputManager.PropagateMouseDown (System.Collections.Generic.IEnumerable`1[T] drawables, osu.Framework.Input.InputState state, osu.Framework.Input.MouseDownEventArgs args) [0x00015] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Input/InputManager.cs:552 
  at osu.Framework.Input.InputManager.handleMouseDown (osu.Framework.Input.InputState state, OpenTK.Input.MouseButton button) [0x0001f] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Input/InputManager.cs:510 
  at osu.Framework.Input.InputManager.updateMouseEvents (osu.Framework.Input.InputState state) [0x0009f] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Input/InputManager.cs:446 
  at osu.Framework.Input.InputManager.HandleNewState (osu.Framework.Input.InputState state) [0x00144] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Input/InputManager.cs:274 
  at osu.Game.Rulesets.UI.RulesetInputManager`1[T].HandleNewState (osu.Framework.Input.InputState state) [0x00001] in /Users/smoogipoo/Repos/Temp/temp-osu/osu.Game/Rulesets/UI/RulesetInputManager.cs:47 
  at osu.Framework.Input.InputManager.Update () [0x00045] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Input/InputManager.cs:217 
  at osu.Game.Rulesets.UI.RulesetInputManager`1[T].Update () [0x000da] in /Users/smoogipoo/Repos/Temp/temp-osu/osu.Game/Rulesets/UI/RulesetInputManager.cs:177 
  at osu.Framework.Graphics.Drawable.UpdateSubTree () [0x000b9] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Graphics/Drawable.cs:352 
  at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree () [0x00001] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Graphics/Containers/CompositeDrawable.cs:467 
  at osu.Game.Rulesets.UI.RulesetInputManager`1[T].UpdateSubTree () [0x00013] in /Users/smoogipoo/Repos/Temp/temp-osu/osu.Game/Rulesets/UI/RulesetInputManager.cs:139 
  at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree () [0x00061] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Graphics/Containers/CompositeDrawable.cs:484 
  at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree () [0x00061] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Graphics/Containers/CompositeDrawable.cs:484 
  at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree () [0x00061] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Graphics/Containers/CompositeDrawable.cs:484 
  at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree () [0x00061] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Graphics/Containers/CompositeDrawable.cs:484 
  at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree () [0x00061] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Graphics/Containers/CompositeDrawable.cs:484 
  at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree () [0x00061] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Graphics/Containers/CompositeDrawable.cs:484 
  at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree () [0x00061] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Graphics/Containers/CompositeDrawable.cs:484 
  at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree () [0x00061] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Graphics/Containers/CompositeDrawable.cs:484 
  at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree () [0x00061] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Graphics/Containers/CompositeDrawable.cs:484 
  at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree () [0x00061] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Graphics/Containers/CompositeDrawable.cs:484 
  at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree () [0x00061] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Graphics/Containers/CompositeDrawable.cs:484 
  at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree () [0x00061] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Graphics/Containers/CompositeDrawable.cs:484 
  at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree () [0x00061] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Graphics/Containers/CompositeDrawable.cs:484 
  at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree () [0x00061] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Graphics/Containers/CompositeDrawable.cs:484 
  at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree () [0x00061] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Graphics/Containers/CompositeDrawable.cs:484 
  at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree () [0x00061] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Graphics/Containers/CompositeDrawable.cs:484 
  at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree () [0x00061] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Graphics/Containers/CompositeDrawable.cs:484 
  at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree () [0x00061] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Graphics/Containers/CompositeDrawable.cs:484 
  at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree () [0x00061] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Graphics/Containers/CompositeDrawable.cs:484 
  at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree () [0x00061] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Graphics/Containers/CompositeDrawable.cs:484 
  at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree () [0x00061] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Graphics/Containers/CompositeDrawable.cs:484 
  at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree () [0x00061] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Graphics/Containers/CompositeDrawable.cs:484 
  at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree () [0x00061] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Graphics/Containers/CompositeDrawable.cs:484 
  at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree () [0x00061] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Graphics/Containers/CompositeDrawable.cs:484 
  at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree () [0x00061] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Graphics/Containers/CompositeDrawable.cs:484 
  at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree () [0x00061] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Graphics/Containers/CompositeDrawable.cs:484 
  at osu.Framework.Platform.GameHost.UpdateFrame () [0x000b2] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Platform/GameHost.cs:250 
  at osu.Framework.Threading.GameThread.ProcessFrame () [0x00040] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Threading/GameThread.cs:115 
  at osu.Framework.Threading.GameThread.runWork () [0x0002d] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Threading/GameThread.cs:104 
  at System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) [0x00014] in /Users/builder/data/lanes/4992/mono-mac-sdk/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/thread.cs:68 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00071] in /Users/builder/data/lanes/4992/mono-mac-sdk/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:957 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in /Users/builder/data/lanes/4992/mono-mac-sdk/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:904 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) [0x0002b] in /Users/builder/data/lanes/4992/mono-mac-sdk/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:893 
  at System.Threading.ThreadHelper.ThreadStart () [0x00008] in /Users/builder/data/lanes/4992/mono-mac-sdk/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/thread.cs:105 
```

To:
```
ERROR] FATAL UNHANDLED EXCEPTION: System.Exception: Try find me!
  at osu.Game.Rulesets.Osu.Objects.Drawables.Pieces.CirclePiece.OnPressed (osu.Game.Rulesets.Osu.OsuAction action) [0x00001] in /Users/smoogipoo/Repos/Temp/temp-osu/osu.Game.Rulesets.Osu/Objects/Drawables/Pieces/CirclePiece.cs:54 
  at osu.Framework.Input.Bindings.KeyBindingContainer`1+<>c__DisplayClass19_1[T].<PropagatePressed>b__1 (osu.Framework.Input.Bindings.IKeyBindingHandler`1[T] d) [0x00000] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Input/Bindings/KeyBindingContainer.cs:156 
  at System.Linq.Enumerable.TryGetFirst[TSource] (System.Collections.Generic.IEnumerable`1[T] source, System.Func`2[T,TResult] predicate, System.Boolean& found) [0x0003f] in /Users/builder/data/lanes/4992/mono-mac-sdk/external/bockbuild/builds/mono-x64/external/corefx/src/System.Linq/src/System/Linq/First.cs:107 
  at System.Linq.Enumerable.FirstOrDefault[TSource] (System.Collections.Generic.IEnumerable`1[T] source, System.Func`2[T,TResult] predicate) [0x00000] in /Users/builder/data/lanes/4992/mono-mac-sdk/external/bockbuild/builds/mono-x64/external/corefx/src/System.Linq/src/System/Linq/First.cs:46 
  at osu.Framework.Input.Bindings.KeyBindingContainer`1[T].PropagatePressed (System.Collections.Generic.IEnumerable`1[T] drawables, T pressed) [0x000bb] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Input/Bindings/KeyBindingContainer.cs:156 
  at osu.Framework.Input.Bindings.KeyBindingContainer`1[T].handleNewPressed (osu.Framework.Input.InputState state, osu.Framework.Input.Bindings.InputKey newKey, System.Boolean repeat) [0x000c8] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Input/Bindings/KeyBindingContainer.cs:129 
  at osu.Framework.Input.Bindings.KeyBindingContainer`1[T].OnMouseDown (osu.Framework.Input.InputState state, osu.Framework.Input.MouseDownEventArgs args) [0x00000] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Input/Bindings/KeyBindingContainer.cs:88 
  at osu.Framework.Graphics.Drawable.TriggerOnMouseDown (osu.Framework.Input.InputState screenSpaceState, osu.Framework.Input.MouseDownEventArgs args) [0x00000] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Graphics/Drawable.cs:1662 
  at osu.Framework.Input.InputManager+<>c__DisplayClass51_0.<PropagateMouseDown>b__0 (osu.Framework.Graphics.Drawable target) [0x00000] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Input/InputManager.cs:552 
  at System.Linq.Enumerable.TryGetFirst[TSource] (System.Collections.Generic.IEnumerable`1[T] source, System.Func`2[T,TResult] predicate, System.Boolean& found) [0x0003f] in /Users/builder/data/lanes/4992/mono-mac-sdk/external/bockbuild/builds/mono-x64/external/corefx/src/System.Linq/src/System/Linq/First.cs:107 
  at System.Linq.Enumerable.FirstOrDefault[TSource] (System.Collections.Generic.IEnumerable`1[T] source, System.Func`2[T,TResult] predicate) [0x00000] in /Users/builder/data/lanes/4992/mono-mac-sdk/external/bockbuild/builds/mono-x64/external/corefx/src/System.Linq/src/System/Linq/First.cs:46 
  at osu.Framework.Input.InputManager.PropagateMouseDown (System.Collections.Generic.IEnumerable`1[T] drawables, osu.Framework.Input.InputState state, osu.Framework.Input.MouseDownEventArgs args) [0x00015] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Input/InputManager.cs:552 
  at osu.Framework.Input.InputManager.handleMouseDown (osu.Framework.Input.InputState state, OpenTK.Input.MouseButton button) [0x0001f] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Input/InputManager.cs:510 
  at osu.Framework.Input.InputManager.updateMouseEvents (osu.Framework.Input.InputState state) [0x0009f] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Input/InputManager.cs:446 
  at osu.Framework.Input.InputManager.HandleNewState (osu.Framework.Input.InputState state) [0x00144] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Input/InputManager.cs:274 
  at osu.Game.Rulesets.UI.RulesetInputManager`1[T].HandleNewState (osu.Framework.Input.InputState state) [0x00001] in /Users/smoogipoo/Repos/Temp/temp-osu/osu.Game/Rulesets/UI/RulesetInputManager.cs:47 
  at osu.Framework.Input.InputManager.Update () [0x00045] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Input/InputManager.cs:217 
  at osu.Game.Rulesets.UI.RulesetInputManager`1[T].Update () [0x000da] in /Users/smoogipoo/Repos/Temp/temp-osu/osu.Game/Rulesets/UI/RulesetInputManager.cs:177 
  at osu.Framework.Graphics.Drawable.UpdateSubTree () [0x000b9] in /Users/smoogipoo/Repos/Temp/temp-osu/osu-framework/osu.Framework/Graphics/Drawable.cs:352 
  at osu.Game.Rulesets.Osu.UI.OsuRulesetContainer.UpdateSubTree ()
  at osu.Game.Screens.Play.PauseContainer.UpdateSubTree ()
  at osu.Framework.Screens.Screen+ContentContainer.UpdateSubTree ()
  at osu.Game.Screens.Play.Player.UpdateSubTree ()
  at osu.Game.Tests.Visual.ScreenTestCase+TestOsuScreen.UpdateSubTree ()
  at osu.Game.Tests.Visual.TestCaseAllPlayers.UpdateSubTree ()
  at osu.Framework.Graphics.Containers.DelayedLoadWrapper.UpdateSubTree ()
  at osu.Framework.Screens.Screen+ContentContainer.UpdateSubTree ()
  at osu.Framework.Testing.TestBrowser.UpdateSubTree ()
  at osu.Game.Graphics.Cursor.OsuTooltipContainer.UpdateSubTree ()
  at osu.Game.Input.Bindings.GlobalKeyBindingInputManager.UpdateSubTree ()
  at osu.Framework.Graphics.Containers.DrawSizePreservingFillContainer.UpdateSubTree ()
  at osu.Desktop.OsuTestBrowser.UpdateSubTree ()
  at osu.Framework.Input.FrameworkActionContainer.UpdateSubTree ()
  at osu.Framework.Input.PlatformActionContainer.UpdateSubTree ()
  at osu.Framework.Input.UserInputManager.UpdateSubTree ()
```

### CODE REVIEW

File: osu.Framework/Graphics/UpdateSubTreeException.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+{
+    public class UpdateSubTreeException : Exception
+    {
+        /// <summary>
+        /// Types that are ignored for the custom stack traces.
+        /// </summary>
+        private static readonly Type[] blacklist =
+        {
+            typeof(Container),
+            typeof(Container<>),
```
Comment (peppy) [inline osu.Framework/Graphics/UpdateSubTreeException.cs]: hmm, not sure about `Container<>` as an exception

**Comment by peppy** [comment]
Looks good to me. @Tom94 would appreciate your opinion on this too!

File: osu.Framework/Graphics/Containers/CompositeDrawable.cs [hunk 2]
```diff
@@ -462,6 +462,8 @@ public override void UpdateClock(IFrameBasedClock clock)
         /// </summary>
         protected virtual bool RequiresChildrenUpdate => !IsMaskedAway || !childrenSizeDependencies.IsValid;
 
+        internal int CurrentUpdateIndex { get; private set; } = -1;
```
Comment (Tom94) [inline osu.Framework/Graphics/Containers/CompositeDrawable.cs]: Why not `CurrentUpdateDrawable`?

**Comment by Tom94** [comment]
Unsure what to think about this. It seems to me like all this does in you example is cut away the bottom, including even the non-UpdateSubTree stuff there. Don't we want to keep that?

**Comment by peppy** [comment]
It's always going to be the same thing though, so not sure we do want to see it. I'm pretty neutral.

**Comment by smoogipoo** [comment]
Cutting out the bottom is more a consequence of this, but the main part of it was giving the type of the intermediate CompositeDrawables that makes the hierarchy easier to follow.

**Comment by Tom94** [comment]
@peppy I agree that most of the time we don't want to see the bottom, but I'd rather leave the stack trace complete. It's not difficult after all to focus on just the relevant part.
@smoogipoo I agree that showing the types is very nice, but I would prefer it the rest of the stack trace would remain intact.

**Comment by Tom94** [comment]
Phoning in @default0 in case he has an opinion on this.

**Comment by default0** [comment]
I'm honestly pretty indifferent about this. It will have weird side effects like not using the class the UpdateSubTree method was called in as the type in the stacktrace, which could be both helpful or irritating, depending on if you are aware of this happening or not and as you already discussed, leaves out the bottom of the stacktrace (and line numbers in the UpdateSubTree-portion of the stacktrace).

Overall I don't see how this improves the debugging experience, but if you find it helpful for your purposes I also don't really see anything wrong with it.

**Comment by peppy** [comment]
I'd say we should not truncate for the time being. I think it may alienate new developers who have no idea what is going on.

**Comment by smoogipoo** [comment]
Let's not do this for the time being then. I think the current output is fine as it is upon a further look.

**Pull request closed without merging**